### ISSUE TITLE
PIDloop's raw DTerm differs from classic PID's error DTerm. This needs to be well signposted

### ISSUE THREAD

**Issue by Dunbaratu** [KSP-KOS/KOS#2693]
kOS's PIDLoop calculates the Dterm based on delta of raw input, not delta of error of input like classic PIDs are described to do.  This becomes relevent when the setpoint changes and users should have the difference very well signposted as it does matter.

The difference is explained below.

In a classic PID as textbooks explain them, the D term is like so:
```
delta_error = (cur_input_error) - (prev_input_error) / deltaT;
Dterm = Kd * delta_error;
```
But in kOS, the C# code running PIDLoop:Update() does this to calulcate the D term:
```
delta_raw = (cur_input_raw) - (prev_input_raw) / deltaT;
Dterm = Kd * delta_raw;
```
Most of the time this makes no difference, since you calculate the Dterm from the change to the value, not its absolute amount, so shifting it up or down by setpoint shouldn't change anything.  i.e. the following would be identical most of the time:

**Version (A):**

when initializing: ``set myPid to PIDLOOP(1, 0.2, 0.02, -1, 1).``
when updating: ``set ctrl to myPid:UPDATE(time:seconds, measurement - wanted).``

**Version (B):**

when initializing: ``set myPid to PIDLOOP(1, 0.2, 0.02, -1, 1).  set myPid:SETPOINT to wanted.``
when updating: ``set ctrl to myPid:UPDATE(time:seconds, measurement).``

Someone who expected kOS's PIDLoop to behave exactly like the textbook case would think the above two are exactly identical, because with a textbook PID, they would be.  But with kOS's PID there is a subtle difference that occurs when your setpoint suddenly changes.  Let's say you were flying a plane with desired descent rate of -4 m/s as your setpoint, and then a new waypoint is picked and the new desired descent rate is -6 m/s.  With the textbook PID, that new update tick will have a massive jump in Dterm for a moment because all of a sudden your error has jumped by 2 m/s because your descent is still -4 m/s but now you want -6 where before you wanted -4.  The system thinks you need to massively jam the controls downward because it suddenly thinks you're accelerating upward at 2 m/s.  And that will happen whether you use version (A) or (B) above.

But with kOS's PIDLoop, that is NOT true.  With kOS's PIDLoop, version (A) will work like the textbook exampe, but (B) will work better.  With kOS, version (B) will lack the sudden phantom jump in Dterm because it's taking the Dterm from raw measurement, not from error of measurement from setpoint.  The error changed suddenly only because the setpoint changed, not because the raw value did.  So it doesn't get a sudden phantom need to "counter" the fast delta that just happened.

tl;dr - with kOS's PIDloop, using a nonzero setpoint (version (B)) instead of subtracting the setpoint yourself  to center the argument around zero (version (A)) will give smoother results when the setpoint gets changed, but users won't know that if they assume it works like the textbook PID examples.  Thus the fact that kOS works this way needs to be better signposted in the docs.  The docs do show the algorithm, but it's easy to miss this key difference between what that pseudocode says and what standard PID's do.
